sql/core/src/main/scala/org/apache/spark/sql/streaming/StreamingQueryListener.scala

```diff
@@ -82,15 +82,15 @@ object StreamingQueryListener {
    * @param id A unique query id that persists across restarts. See `StreamingQuery.id()`.
    * @param runId A query id that is unique for every start/restart. See `StreamingQuery.runId()`.
    * @param name User-specified name of the query, null if not specified.
-   * @param submissionTime The timestamp to start a query.
+   * @param timestamp The timestamp to start a query.
    * @since 2.1.0
    */
   @Evolving
   class QueryStartedEvent private[sql](
       val id: UUID,
       val runId: UUID,
       val name: String,
-      val submissionTime: Long) extends Event
+      val timestamp: String) extends Event
```

Comment:
This changes made harder processing of event logs machinery, for instance by Spark. If you have at least the following two records:
```
{"Event":"SparkListenerExecutorAdded","Timestamp":1603819505374, ...}
{"Event":"org.apache.spark.sql.streaming.StreamingQueryListener$QueryStartedEvent","timestamp":"2020-10-27T17:37:48.452Z" ...}
``` 
and want to load them by Spark. It will fails with:
```
Found duplicate column(s) in the data schema: `timestamp`;
org.apache.spark.sql.AnalysisException: Found duplicate column(s) in the data schema: `timestamp`;
	at org.apache.spark.sql.util.SchemaUtils$.checkColumnNameDuplication(SchemaUtils.scala:90)
	at org.apache.spark.sql.util.SchemaUtils$.checkSchemaColumnNameDuplication(SchemaUtils.scala:45)
```
because we have two columns with the same name `timestamp` but with different types.

How about to unify schemas of all events (at least for top level columns), and use the same names and types across all `SparkListenerEvent`?

Reply:
+1. Which type shall we use? String or Long?

Reply:
To be compatible to the events from `JsonProtocol`:
- SparkListenerJobStart
- SparkListenerJobEnd
- SparkListenerBlockManagerAdded
- SparkListenerBlockManagerRemoved
- SparkListenerExecutorAdded
- SparkListenerExecutorRemoved
- SparkListenerExecutorBlacklisted
- SparkListenerExecutorBlacklistedForStage
- SparkListenerNodeBlacklistedForStage
- SparkListenerExecutorUnblacklisted
- SparkListenerNodeBlacklisted
- SparkListenerNodeUnblacklisted
- SparkListenerApplicationStart
- SparkListenerApplicationEnd

It should be `Long`.

Reply:
as an option if we want to have human-readable timestamps and avoid conflicts with other events, we could rename:
`timestamp: String` -> `submissionTime: String`

Reply:
This is a public API. Renaming it would break compatibility.

The consumer of Spark event logs is Spark History Server. It's not designed to be friendly to Spark JSON data source since the beginning. If we would like to make Spark JSON source easy to read event logs, it would need a new design. Currently, all fields of each Spark event class is in the top level column and these fields are arbitrary, and we don't have enforcement on these event classes. Delta Lake has some similar classes that need to be serialized to JSON. We create a wrapper [SingleAction](https://github.com/delta-io/delta/blob/v0.7.0/src/main/scala/org/apache/spark/sql/delta/actions/actions.scala#L344) to store all possible types of `Action`s. This may be a possible approach for new Spark event logs. However, maintaining the compatibility is hard. This is probably not worth.